atomisticparsers/gromacs/parser.py

```diff
@@ -640,6 +673,42 @@ def __init__(self):
         ]
         super().__init__()
 
+    def get_pbc(self):
+        pbc = self.input_parameters.get("pbc", "xyz")
+        return ["x" in pbc, "y" in pbc, "z" in pbc]
+
+    def get_mdp_file(self):
+        """
+        Tries to find the mdp input parameters (ext = mdp) that match the mainfile calculation.
+        Priority is as follows:
+            1. output mdp file containing both the matching mainfile name and the standard
+            gromacs name `mdout`
+            2. file containing the standard gromacs name `mdout`
+            3. input mdp file matching the mainfile name (as usual)
+            4. any `.mdp` file within the directory (as usual)
+        """
+        ext = "mdp"
+        std_filename = "mdout"
+        files = [d for d in self._gromacs_files if d.endswith(ext)]
+
+        if len(files) == 0:
+            return ""
+
+        if len(files) == 1:
+            return os.path.join(self._maindir, files[0])
+
+        for f in files:
+            filename = f.rsplit(".", 1)[0]
+            if self._basename in filename and std_filename in filename:
+                return os.path.join(self._maindir, f)
+
+        for f in files:
+            filename = f.rsplit(".", 1)[0]
+            if std_filename in filename:
```

Comment:
put this in the other loop?

Reply:
I was trying to prioritize the situation where someone incorporated both the basename and std_filename into the file name, e.g., `<basename>_<std_filename>.mdp`. But it is possible that `<basename>_<std_filename>.mdp` and `<std_filename>.mdp` both correspond to files in the directory. I don't see any way to easily use 1 loop for this, do you?

Reply:
ah i understand then it is fine.